core/course_controller/CourseController.py

```diff
@@ -2,10 +2,30 @@
 
 from core.local_data_classes import CourseFormData, CourseOverview, CourseRef, UserRef, CourseSectionRef, LabSectionRef
 from django.db import models
-from ta_scheduler.models import Course, CourseSection, LabSection
+from ta_scheduler.models import Course, CourseSection, LabSection, Semester
+
 
 
 class CourseController:
+
+    @staticmethod
+    def has_duplicate_course_id(course_code: str, semester: Semester, exclude_course_id: int | None = None) -> bool:
```

Comment:
This method should not be necessary for the course tests. Use error handling from the get method for this case

Reply:
for the course form *